nibabel/filebasedimages.py

```diff
@@ -252,10 +252,12 @@ def set_filename(self, filename):
            ``.file_map`` attribute. Otherwise, the image instance will
            try and guess the other filenames from this given filename.
         '''
+        filename = _stringify_path(filename)
```

Comment:
Working on this now, but had a question since `filespec_to_file_map` calls `types_filename`, do you think it would be safe to only put the guard in `types_filename` and not in `filespec_to_file_map` at all?

Reply:
That seems reasonable.